File: 백준/Silver/2559. 수열/수열.py
import sys

n,k = map(int, sys.stdin.readline().rstrip().split())
arr = list(map(int, sys.stdin.readline().rstrip().split()))

prefix_sum = [0] * n

prefix_sum[0] = arr[0]

for i in range(1,n):
    prefix_sum[i] = prefix_sum[i-1] + arr[i]

ans = []

for i in range(k-1,n):
    if i == k-1:
        ans.append(prefix_sum[i])
    else:
        ans.append(prefix_sum[i] - prefix_sum[i-k])
print(max(ans))

File: 백준/Silver/2559. 수열/수열.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        List<Integer> ans = new ArrayList<>();

        int[] arr = new int[n];
        int[] prefix = new int[n + 1];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        for (int i = 0; i < n; i++) {
            prefix[i+1] = prefix[i] + arr[i];
        }

        for (int i = k; i <= n; i++) {
            ans.add(prefix[i] - prefix[i-k]);
        }

        System.out.println(Collections.max(ans));








    }
}